### PR TITLE
Fixing docs.yml to point to missing pages

### PR DESCRIPTION
**Pull request by manez**
**GitHub Issue**: https://github.com/Islandora/documentation/issues/1352

# What does this Pull Request do?

Updates mkdcos.yml to include a number of pages that were missing from the index but exist in our docs folder. Rearranged something under grouped headings ot try to make the docs more navigable. 

As a bonus, added a missing screenshot from the Collections documentation.

# What's new?
- A screenshot on the Collections docs (asset was missing)
- Links to:
      - alpaca/alpaca-technical-stack.md
      - contributing/hacking-on-islandora.md
      - islandora/drupal-bundle-configurations.md
      - technical-documentation/checking-coding-standards.md
      - technical-documentation/drupal-project.md
      - technical-documentation/resizing_vm.md
      - user-documentation/intro-to-ld-for-islandora-8.md
      - user-documentation/versioning.md 
- New subheadings to group similar pages in the docs:
   - Content in Islandora 8
   - How-To
   - Integrations
   - Tests
   - Alpaca
   - Virtual Machines

# How should this be tested?

Look at the new docs build and verify that links go to the things they should go to and I haven't broken anything.

# Additional Notes:
There are a number of pages from the list in #1352 that I haven't added here because they don't really have a place in the official docs. I'll address those in a separate pull request.

# Interested parties
@Islandora/8-x-committers

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -70,21 +76,28 @@ nav:
...
     - 'Flysystem': 'technical-documentation/flysystem.md'
     - 'JWT Authentication': 'technical-documentation/jwt.md'
     - 'Versioning Policy': 'technical-documentation/versioning.md'
     - 'How to Build Documentation': 'technical-documentation/docs-build.md'
-    - 'Testing Notes': 'technical-documentation/testing-notes.md'
     - 'Adding back ?_format=jsonld': 'technical-documentation/adding_format_jsonld.md'
     - 'Updating a `deb` and adding it to Lyrasis PPA': 'technical-documentation/ppa-documentation.md'
-    - 'Alpaca Tips': 'technical-documentation/alpaca_tips.md'
-
+    - Aplaca:
```
Comment (whikloj) [inline mkdocs.yml]: Typo

**Review by whikloj — CHANGES_REQUESTED** [review]
Also, we should probably look if these pages are at all relevant anymore. That can be a separate issue but I just don't know if all these pages we are exposing are still correct.

**Comment by whikloj** [comment]
And now I am seeing https://github.com/Islandora/documentation/pull/1421 which means you might have already reviewed these pages.

**Comment by manez** [comment]
I gave them a light review and addressed everything that seemed clearly outdated or irrelevant with #1421, but they should definitely get a closer look next time we do a full doc review.

**Comment by manez** [comment]
~Except https://github.com/Islandora/documentation/blob/master/docs/contributing/hacking-on-islandora.md, which will get its own PR to get the CLAWs out.~

This one is beyond me. It's more than just a de-CLAWing.

**Comment by whikloj** [comment]
@manez you'll have to appease the "Making a git merge useless" check by pressing the button 🤦‍♂

**Review by whikloj — APPROVED** [review]

**Pull request merged**